include/pmacc/algorithms/math/floatMath/bessel.tpp

```diff
@@ -0,0 +1,196 @@
+/* Copyright 2016-2017 Alexander Debus
+ *
+ * This file is part of PMacc.
+ *
+ * PMacc is free software: you can redistribute it and/or modify
+ * it under the terms of either the GNU General Public License or
+ * the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PMacc is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License and the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * and the GNU Lesser General Public License along with PMacc.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "pmacc/types.hpp"
+#include <boost/math/special_functions/bessel.hpp>
+
+
+namespace pmacc
+{
+namespace algorithms
+{
+namespace math
+{
+namespace bessel
+{
+
+    template< >
+    struct I0< float >
+    {
+        using result = float;
+
+        HDINLINE result operator( )( result const & x )
+        {
+#if __CUDA_ARCH__
+            return ::cyl_bessel_i0f( x );
+#else
+            return boost::math::cyl_bessel_i(
+                0,
+                x
+            );
+#endif
+        }
+    };
+
+    template< >
+    struct I1< float >
+    {
+        using result = float;
+
+        HDINLINE result operator( )( result const & x )
+        {
+#if __CUDA_ARCH__
+            return ::cyl_bessel_i1f( x );
+#else
+            return boost::math::cyl_bessel_i(
+                1,
+                x
+            );
+#endif
+        }
+    };
+
+    template< >
+    struct J0<float>
+    {
+        using result = float;
+
+        HDINLINE result operator( )( result const & x )
+        {
+#if __CUDA_ARCH__
+            return ::j0f( x );
+#else
+            return boost::math::cyl_bessel_j(
+                0,
+                x
+            );
+#endif
+        }
+    };
+
+    template< >
+    struct J1<float>
+    {
+        using result = float;
+
+        HDINLINE result operator( )( result const & x )
+        {
+#if __CUDA_ARCH__
+            return ::j1f( x );
+#else
+            return boost::math::cyl_bessel_j(
+                1,
+                x
+            );
+#endif
+        }
+    };
+
+    template< >
+    struct Jn<int, float>
```

Comment:
- missing empty space after `<` and before `>`
- please use one line per argument